BEGINNERS.md

```diff
@@ -0,0 +1,309 @@
+
+If there are any failures either when compiling or testing,
+and you don't know how to fix the issue or don't understand the error,
+ask on the [IRC channel](https://snowdrift.coop/p/snowdrift/w/en/irc),
+and someone will probably help you.
+
+### Committing your changes
+
+When your updates all compile, tests pass,
+and you are ready to submit to the main Snowdrift project,
+you can *commit* your changes.
+
+If you don't already know git basics, we urge you to learn
+about the process of staging with `git add` before committing
+and build good review habits with tools like `git status` and `git diff`.
+See the links for learning resources at the end of this file.
+
+For now, though it isn't best practice, you can quickly commit
+all your changes with the command:
+
+    git commit -a
+
+An editor will show asking you to summarize your changes.
+Make the message one that will be meaningful to people skimming
+all the commits in the future. Then save and close the editor.
+For reference, here's a decent guide to
+[writing good commit messages](https://github.com/erlang/otp/wiki/Writing-good-commit-messages).
+
+## Getting your changes merged
+
+After committing, send your changes to your github account with:
+
+    git push -u my-snow some_branch
+
+Change `some_branch` to the name of the branch where you made the
+commit(s).
+
+Reload the GitHub page with your fork.
+You should see something like this:
+
+![](https://a.pomf.se/paqzzx.png)
+    
+Click the "Pull request" button. There'll be a nice little form there
+for submitting the pull request. Fill out the form, and send it.
+
+Someone should comment on it soon (hopefully within a few hours).
+
+
+## Learning resources and helpful tools
+
+Besides reading the content on the Snowdrift.coop site itself,
+we suggest reviewing our [GUIDE.md](GUIDE.md), specifically sections on
+"Development guidelines and notes" and "Additional notes about databases".
+
+If you want a deeper understanding of various elements in our development,
+here are some resources:
+
+*   The following WikiBooks are fully FLO and include links to
+    further resources as well. As they are wikis, you can and should
+    *improve* them further yourself as you read!
+
+    + [HTML Wikibook](https://en.wikibooks.org/wiki/HyperText_Markup_Language)
+    + [CSS Wikibook](https://en.wikibooks.org/wiki/Cascading_Style_Sheets)
+    + [JavaScript Wikibook](https://en.wikibooks.org/wiki/JavaScript)
+    + [Haskell Wikibook](https://en.wikibooks.org/wiki/Haskell)
+    + [SQL Wikibook](https://en.wikibooks.org/wiki/Structured_Query_Language)
+      is a complete book and good general overview
+    + [A Quick Introduction to Unix](https://en.wikibooks.org/wiki/A_Quick_Introduction_to_Unix)
+      is a good overview to command-line and Unix basics
+    + [Git Wikibook](https://en.wikibooks.org/wiki/Git) is only half-done
+      but has some useful bits
+
+*   [w3schools](http://www.w3schools.com/) is a very good way to learn
```

Comment:
w3schools is the *worst* resource on the internet, and should be banned, not reccomended.

Reccomend something like the Mozilla Developer Network instead